cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -53,6 +53,7 @@ void column_buffer::create(size_type _size,
     _null_mask =
       cudf::detail::create_null_mask(size, mask_state::ALL_NULL, rmm::cuda_stream_view(stream), mr);
   }
+  this->mr = mr;
```

Comment:
I think this still needs more work: The line 39 above creates a buffer without using `mr`.

Reply:
IIUC `_strings` is not part of the "output" data so it shouldn't use `mr`.

Reply:
`_strings` may be part of the output:
```
make_column(...) {
...
  return make_strings_column(*buffer._strings, stream, mr);
}
```

Reply:
Humnn, it seems that I was wrong. `make_strings_column` copies from `_strings` to the output.

Reply:
`string_index_pair` is a pair of char pointer and size which are information used to construct the output data but they are not output data themselves.

Reply:
Correct, the fact that `make_strings_column` makes a copy is precisely why this code needs to be specialized in this way for strings. Since it makes a copy, strings end up being the only branch where the first allocation is a temporary. This is why strings here don't use `mr` in `create` but then need to use it later, whereas all other types move from the buffers constructed in `create` so those buffers need to use `mr`.

Reply:
Thanks all.
@vyasr Can you add comments clarifying this into the code for creating `_strings`? So other devs will figure this out easier next time.

Reply:
Good call, yes I can do that.